sphinx/ext/autodoc/__init__.py

```diff
@@ -1175,22 +1175,73 @@ def format_args(self, **kwargs: Any) -> str:
         if self.env.config.autodoc_typehints == 'none':
             kwargs.setdefault('show_annotation', False)
 
-        # for classes, the relevant signature is the __init__ method's
-        initmeth = self.get_attr(self.object, '__init__', None)
-        # classes without __init__ method, default __init__ or
-        # __init__ written in C?
-        if initmeth is None or \
-                inspect.is_builtin_class_method(self.object, '__init__') or \
-                not(inspect.ismethod(initmeth) or inspect.isfunction(initmeth)):
-            return None
-        try:
-            self.env.app.emit('autodoc-before-process-signature', initmeth, True)
-            sig = inspect.signature(initmeth, bound_method=True)
+        def get_user_defined(obj, attr):
+            if inspect.is_builtin_class_method(obj, attr):
+                return None
+            attr = self.get_attr(obj, attr, None)
+            if not (inspect.ismethod(attr) or inspect.isfunction(attr)):
+                return None
+            return attr
+
+        sig = None
+
+        # this sequence is copied from inspect._signature_from_callable
+
+        # First, let's see if it has an overloaded __call__ defined
+        # in its metaclass
+        if sig is None:
+            call = get_user_defined(type(self.object), '__call__')
+
+            if call is not None:
+                # types which have confusing metaclass signatures it would be best
+                # not to show
+                call_blacklist = [
+                    'enum.EnumMeta.__call__',
+                ]
+                if "{0.__module__}.{0.__qualname__}".format(call) in call_blacklist:
+                    call = None
+
+            if call is not None:
+                self.env.app.emit('autodoc-before-process-signature', call, True)
+                try:
+                    sig = inspect.signature(call, bound_method=True)
+                except TypeError:
+                    print("Faild, oh no")
```

Comment:
This must be updated. Should this be replaced with a warning? Or be suppressed?

Reply:
Whoops, this was a debug statement!

Reply:
Be suppressed, because that's what this code did before.